apps/studio/src/contexts/EditorDrawerContext.tsx

```diff
@@ -1,6 +1,7 @@
 import type { IsomerSchema } from "@opengovsg/isomer-components"
 import type { Dispatch, PropsWithChildren, SetStateAction } from "react"
-import { createContext, useContext, useState } from "react"
+import { createContext, useCallback, useContext, useState } from "react"
+import { flushSync } from "react-dom"
```

Comment:
hmm, a bit concerned that the docs mention:

> Using flushSync is uncommon and can hurt the performance of your app.

> Most of the time, flushSync can be avoided. Use flushSync as last resort.

Actually thought this could be just a debounce issue. Was this the only solution of the alternatives mentioned?

Reply:
yea this was the only thing that worked :sadge: